src/components/Recipe/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { createPortal } from 'react-dom'
+
+import { fetchRecipe } from 'api'
+import Loader from 'components/common/Loader'
+import { Ingredient, RecipeDetails, Step } from 'types'
+
+import styles from './Recipe.module.scss'
+
+function Recipe() {
+  const { id } = useParams()
+
+  const [recipe, setRecipe] = useState<RecipeDetails>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    fetchRecipe(id!).then((response) => {
+      setRecipe(response.data)
+      setIsLoading(false)
+    })
+  }, [id])
+
+  return (
+    <>
+      {isLoading && createPortal(<Loader />, document.body)}
+      {recipe && (
+        <div className={styles.container}>
+          <h2 className={styles.title}>{recipe.title}</h2>
+          <img className={styles.photo} src={recipe.image} alt="" />
+          <div className={styles.details}>
+            <div className={styles.ingredients}>
+              <h3 className={styles.ingredients__title}>Ingredients</h3>
+              <ul className={styles.ingredients__list}>
+                {recipe.extendedIngredients.map((ingredient: Ingredient) => {
+                  return (
```

Comment:
could be without return like map((something) => (

but linter should do that job, if it's not screaming with errors than its okay hehe

Reply:
@sasza2 linter is crying that he wants `return` keyword :/